mobile/src/main/java/rs/readahead/washington/mobile/views/activity/VideoViewerActivity.kt

```diff
@@ -0,0 +1,569 @@
+import rs.readahead.washington.mobile.views.fragment.vault.attachements.PICKER_FILE_REQUEST_CODE
+import rs.readahead.washington.mobile.views.fragment.vault.attachements.WRITE_REQUEST_CODE
+import rs.readahead.washington.mobile.views.fragment.vault.info.VaultInfoFragment
+
+@RuntimePermissions
+class VideoViewerActivity : BaseLockActivity(), PlayerControlView.VisibilityListener, StyledPlayerControlView.VisibilityListener,
+    IMediaFileViewerPresenterContract.IView {
+    private lateinit var simpleExoPlayerView: StyledPlayerView
+    private var player: ExoPlayer? = null
+    private lateinit var binding: ActivityVideoViewerBinding
+    private var trackSelector: DefaultTrackSelector? = null
+    private var needRetrySource = false
+    private var shouldAutoPlay = false
+    private var withMetadata = false
+    private var resumeWindow = 0
+    private var resumePosition: Long = 0
+    private var vaultFile: VaultFile? = null
+    private var toolbar: Toolbar? = null
+    private var actionsDisabled = false
+    private var presenter: MediaFileViewerPresenter? = null
+    private var alertDialog: AlertDialog? = null
+    private var progressDialog: ProgressDialog? = null
+    private var isInfoShown = false
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityVideoViewerBinding.inflate(layoutInflater)
+
+        overridePendingTransition(R.anim.slide_in_start, R.anim.fade_out)
+        setContentView(binding.root)
+
+        if (intent.hasExtra(NO_ACTIONS)) {
+            actionsDisabled = true
+        }
+        setupToolbar()
+        shouldAutoPlay = true
+        clearResumePosition()
+        simpleExoPlayerView = findViewById(R.id.player_view)
+        simpleExoPlayerView.setControllerVisibilityListener(this)
+        simpleExoPlayerView.requestFocus()
+        presenter = MediaFileViewerPresenter(this)
+    }
+
+    override fun finish() {
+        super.finish()
+        overridePendingTransition(R.anim.slide_in_end, R.anim.slide_out_start)
+    }
+
+    public override fun onNewIntent(intent: Intent) {
+        super.onNewIntent(intent)
+        releasePlayer()
+        shouldAutoPlay = true
+        clearResumePosition()
+        setIntent(intent)
+    }
+
+    public override fun onStart() {
+        super.onStart()
+        if (SDK_INT > 23) {
+            initializePlayer()
+        }
+    }
+
+    public override fun onResume() {
+        super.onResume()
+        if (SDK_INT <= 23 || player == null) {
+            initializePlayer()
+        }
+    }
+
+    public override fun onPause() {
+        super.onPause()
+        if (SDK_INT <= 23) {
+            releasePlayer()
+        }
+    }
+
+    public override fun onStop() {
+        super.onStop()
+        if (SDK_INT > 23) {
+            releasePlayer()
+        }
+    }
+
+    override fun onDestroy() {
+        if (alertDialog != null && alertDialog!!.isShowing) {
```

Comment:
This will throw a `NullPointerException` if alertDialog is `null`, make sure that alertDialog is not null then check the rest.